### PR TITLE
Split JabRefFrame to View and ViewModel

### PR DESCRIPTION
**Pull request by calixtus**
Plan is to implement the [mvvm pattern](https://devdocs.jabref.org/code-howtos/javafx.html) somehow in JabRefFrame and clean up a bit the gui root directory.

- Split JabRefFrame into JabRefFrame and JabRefFrameViewModel
- TabContainer means the tabbed pane, but it is part of JabRefFrame, so the wrapper methods (addTab close etc) stay there.
- Moved JabRefFrame and the related classes to a distinct package frame
tbc.

Please leave your comments and thoughts.

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by koppor** [comment]
For viewing the diff, I recommend to try out [RefactoringMiner](https://github.com/tsantalis/RefactoringMiner). See https://github.com/tsantalis/RefactoringMiner/blob/master/docker/README.md for a howto.

File: src/main/java/org/jabref/gui/LibraryTabContainer.java [hunk 1]
```diff
@@ -4,10 +4,8 @@
 
 import org.jabref.model.database.BibDatabaseContext;
 
-import org.jspecify.annotations.NullMarked;
 import org.jspecify.annotations.Nullable;
 
-@NullMarked
```
Comment (koppor) [inline src/main/java/org/jabref/gui/LibraryTabContainer.java:10]: Oh, some things can be `null` here?

**Review by koppor — COMMENTED** [review]
Looks good so far

File: src/main/java/org/jabref/gui/importer/actions/OpenDatabaseAction.java [hunk 2]
```diff
@@ -180,7 +181,7 @@ public void openFiles(List<Path> filesToOpen) {
     /**
      * This is the real file opening. Should be called via {@link #openFile(Path)}
      *
-     * Similar method: {@link org.jabref.gui.JabRefFrame#addTab(org.jabref.model.database.BibDatabaseContext, boolean)}.
```
Comment (koppor) [inline src/main/java/org/jabref/gui/importer/actions/OpenDatabaseAction.java:183]: Please do not remove the absolute package for JavaDoc comments. I think, checkstyle will complain or some other checker?

**Comment by calixtus** [comment]
Stopping refactoring here, since the pr has already grown very large and further refactorings require changes to LibraryTab and Globals, which would totally escalate this.

**Review by Siedlerchr — APPROVED** [review]

**Comment by LoayGhreeb** [comment]
@calixtus After this PR, many warnings related to CSS appear when trying to open the entry editor for the first time, but not for all libraries. For example, `Chocolate.bib` always reproduces these warnings, but `jabref-authors.bib` works without any warning.
```
WARN: Caught 'java.lang.ClassCastException: class java.lang.String cannot be cast to class javafx.scene.paint.Paint (java.lang.String is in module java.base of loader 'bootstrap'; javafx.scene.paint.Paint is in module javafx.graphics@22 of loader 'app')' while converting value for '-fx-background-color' from rule '*.editorPane' in stylesheet file:/C:/Users/Loay%20Ahmed/Documents/JabRef/build/resources/main/org/jabref/gui/entryeditor/entryeditor.css
2024-04-17 00:17:00 [JavaFX Application Thread] sun.util.logging.internal.LoggingProviderImpl$JULWrapper.log()
WARN: Could not resolve '-fx-text-background-color' while resolving lookups for '-fx-text-fill' from rule '*.label' in stylesheet jar:file:///C:/Users/Loay%20Ahmed/.gradle/caches/modules-2/files-2.1/org.openjfx/javafx-controls/22/3f2f311e6cff97b2571231e7d9d3acc6ac100d90/javafx-controls-22-win.jar!/com/sun/javafx/scene/control/skin/modena/modena.bss
2024-04-17 00:17:00 [JavaFX Application Thread] sun.util.logging.internal.LoggingProviderImpl$JULWrapper.log()
WARN: Could not resolve '-fx-text-background-color' while resolving lookups for '-fx-text-fill' from rule '*.label' in stylesheet jar:file:///C:/Users/Loay%20Ahmed/.gradle/caches/modules-2/files-2.1/org.openjfx/javafx-controls/22/3f2f311e6cff97b2571231e7d9d3acc6ac100d90/javafx-controls-22-win.jar!/com/sun/javafx/scene/control/skin/modena/modena.bss
2024-04-17 00:17:00 [JavaFX Application Thread] sun.util.logging.internal.LoggingProviderImpl$JULWrapper.log()


And more...
```

I looked into this issue, and the changes were made in the pull request. I tried to revert the commit [`8efbaad` (#11180)](https://github.com/JabRef/jabref/pull/11180/commits/8efbaad269f8e5d294c588713138be230d4817cb), and it works without any warnings.

**Comment by calixtus** [comment]
I see, thing is, the controlsfx decoration pane is made for sthgs completely different purpose (showing another node on top of a label vor validation). So the warnings are showing that there was something wrong before I removed the decoration pane. I will look into it.

**Comment by calixtus** [comment]
I tried, but I cannot reproduce...
Did you try gradle clean?

**Comment by LoayGhreeb** [comment]
> Did you try gradle clean?

tried now, but still able to reproduce

**Comment by calixtus** [comment]
A heisenbug, found a JDK issue:
https://bugs.openjdk.org/browse/JDK-8268657?jql=project%20%3D%20JDK%20AND%20component%20%3D%20javafx%20AND%20text%20~%20%22ClassCastException%20calculateValue%22

**Comment by calixtus** [comment]
Similar here: https://github.com/mkpaz/atlantafx/issues/65#issuecomment-1644214727

**Comment by koppor** [comment]
I searched our issue tracker and found https://github.com/JabRef/jabref/issues/8628. I think, however, the exception is not risen any more and that issue can be closed - and we need a new issue to track?

**Comment by LoayGhreeb** [comment]
In the latest release (5.13), there are no warnings when opening the entry editor. However, upon opening the "Search results" tab, these warnings appear in the event log:

```
Caught 'java.lang.ClassCastException: class java.lang.String cannot be cast to class javafx.scene.paint.Paint (java.lang.String is in module java.base of loader 'bootstrap'; javafx.scene.paint.Paint is in module javafx.graphics@22 of loader 'app')' while converting value for '-fx-background-color' from rule '*.progress-indicator:indeterminate *.segment' in stylesheet jar:file:///C:/Users/Loay%20Ahmed/.gradle/caches/modules-2/files-2.1/org.openjfx/javafx-controls/22/3f2f311e6cff97b2571231e7d9d3acc6ac100d90/javafx-controls-22-win.jar!/com/sun/javafx/scene/control/skin/modena/modena.bss
```
The exceptions mentioned in issue #8628 are not raised. 
 
In the latest main, the same behavior occurs along with the warnings when opening the entry editor.

**Pull request merged**